akka-cluster/src/main/scala/akka/cluster/JoinConfigCompatChecker.scala

```diff
@@ -0,0 +1,166 @@
+package akka.cluster
+
+import java.util
+
+import akka.actor.{ ActorSystem, ExtendedActorSystem }
+import akka.annotation.InternalApi
+import com.typesafe.config.{ Config, ConfigFactory, ConfigUtil, ConfigValue }
+
+import scala.collection.JavaConverters._
+import scala.collection.{ immutable ⇒ im }
+
+trait JoinConfigCompatChecker {
+
+  /** A Set of configuration keys that are required for this checker */
+  def requiredKeys: Set[String]
+
+  /**
+   * Runs the Config check.
+   *
+   * Implementers are free to define what makes Config entry compatible or not.
+   * We do provide some pre-build checks tough: [[JoinConfigCompatChecker.exists()]] and [[JoinConfigCompatChecker.fullMatch()]]
+   *
+   *
+   * @param toCheck - the Config instance to be checked
+   * @param currentConfig - the Config instance containing the expected values
+   * @return a [[ConfigCheck]]. Can be [[Compatible]] or [[Incompatible]], the later must contain a descriptive list of error messages.
+   */
+  def check(toCheck: Config, currentConfig: Config): ConfigCheck
```

Comment:
`ConfigCheck` is marked as internal but here public. It should be public so that external modules or even applications can implement their checkers

Reply:
Yep, I missed this one. Thanks